polaris/hub/client.py

```diff
@@ -520,6 +520,13 @@ def upload_dataset(
                 See also: https://www.python-httpx.org/advanced/#timeout-configuration
             owner: Which Hub user or organization owns the artifact. Takes precedence over `dataset.owner`.
         """
+
+        # Check if the specified license for the dataset is supported
+        if dataset.license and dataset.license.id not in SupportedLicenseType:
```

Comment:
Missed something: The license should always defined when uploading to the hub! Can no longer be None!